pkg/option/config.go

```diff
@@ -2495,7 +2498,8 @@ func (c *DaemonConfig) TunnelDevice() string {
 // if the primary mode is native routing. For example, in the egress gateway,
 // we may send such traffic to a gateway node via a tunnel.
 func (c *DaemonConfig) TunnelExists() bool {
-	return c.TunnelingEnabled() || c.EnableIPv4EgressGateway
+	return c.TunnelingEnabled() || c.EnableIPv4EgressGateway ||
+		(c.NodePortMode == NodePortModeDSR && c.LoadBalancerDSRDispatch == DSRDispatchGeneve)
```

Comment:
Is this function intended to be used with tunneling enabled for DSR? I  don't see the IPIP dispatch case here.

At a glance, it looks like this is used for Pod MTU calculation, but this shouldn't impact Pod MTUs right?

Reply:
(At a glance this code looks correct, I'm just not sure if the code calling this function is accounting for this case)

Reply:
We send packets from intermediate nodes to backend pods through the tunnel. So it affects Pod MTU partially. 

As far as I know, Cilium calculates the MTU considering the tunnel overhead(EthernetMTU - TunnelOverhead) if the tunnel exits, and adds route entries for pod CIDR.
```
root@kind-worker:/# ip r
default via 172.18.0.1 dev eth0
10.0.0.0/24 via 10.0.0.211 dev cilium_host src 10.0.0.211
10.0.0.211 dev cilium_host scope link
10.0.1.0/24 via 10.0.0.211 dev cilium_host src 10.0.0.211 mtu 1450
10.0.2.0/24 via 10.0.0.211 dev cilium_host src 10.0.0.211 mtu 1450
10.0.3.0/24 via 10.0.0.211 dev cilium_host src 10.0.0.211 mtu 1450
172.18.0.0/16 dev eth0 proto kernel scope link src 172.18.0.3
```

Reply:
We've already taken the encap overhead into account for the DSR traffic in the datapath logic. So we might not need to adjust the MTU for the node route.
https://github.com/cilium/cilium/pull/23890/files#diff-304969c111195cf4253b336fc67c9891f17b91d49738a0cc334333ac11e84b9eR278-R281

The DSR logic uses the device MTU (without overhead reduction) for the dsr_is_too_big check.
https://github.com/cilium/cilium/pull/23890/files#diff-304969c111195cf4253b336fc67c9891f17b91d49738a0cc334333ac11e84b9eR278-R281

Reply:
Dropped this change and left a comment. PTAL

Reply:
LGTM, thanks!